engine/src/main/java/org/terasology/config/RenderingConfig.java

```diff
@@ -657,4 +623,74 @@ public void setVolumetricFog(boolean volumetricFog) {
         propertyChangeSupport.firePropertyChange(VOLUMETRIC_FOG, oldValue, this.volumetricFog);
     }
 
+    public void loadDefaultRenderingConfig(FlexibleConfig flexibleConfig) {
+        this.flexibleConfig = flexibleConfig;
+
+        // flexibleConfig.add(new SettingImpl<>(new SimpleUri("engine:pixelFormat"), 24));
+        flexibleConfig.add(new SettingImpl<>(new SimpleUri("engine:windowPosX"), 0));
```

Comment:
Yes, it will be possible to do that. I proposed something similar in the last couple of comments on my PR. This change will also have the added benefit of not requiring the `SettingImpl` class being exposed to the API, satiating @oniatus's concerns. I also think that the `add(Setting setting)` method should entirely be removed to prevent setting publishers from putting in their own potentially malicious `Setting` implementations, but that may be something to discuss at another time.

Reply:
@eviltak: what do you have in mind when you think about a malicious setting implementation?

Regarding the SettingImpl class, I'm not bought on the idea that it shouldn't be public.